tests/integration/src/felt_conversion.rs

```diff
@@ -0,0 +1,117 @@
+use miden_core::Felt;
+use miden_core::StarkField;
+
+/// Wrapper around `Felt` that implements `From` for a bunch of types that are missing in Winterfell
+/// Should be moved to Winterfell eventually
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub struct TestFelt(pub Felt);
+
+impl From<bool> for TestFelt {
+    fn from(b: bool) -> Self {
+        Self(Felt::from(b as u32))
+    }
+}
+
+impl From<u8> for TestFelt {
+    fn from(t: u8) -> Self {
+        Self(t.into())
+    }
+}
+
+impl From<i8> for TestFelt {
+    fn from(t: i8) -> Self {
+        Self((t as u8).into())
+    }
+}
+
+impl From<i16> for TestFelt {
+    fn from(t: i16) -> Self {
+        Self((t as u16).into())
+    }
+}
+
+impl From<u16> for TestFelt {
+    fn from(t: u16) -> Self {
+        Self(t.into())
+    }
+}
+
+impl From<i32> for TestFelt {
+    fn from(t: i32) -> Self {
+        Self((t as u32).into())
+    }
+}
+
+impl From<u32> for TestFelt {
+    fn from(t: u32) -> Self {
+        Self(t.into())
+    }
+}
+
+impl From<u64> for TestFelt {
+    fn from(t: u64) -> Self {
+        Self(t.into())
+    }
+}
+
+impl From<i64> for TestFelt {
+    fn from(t: i64) -> Self {
+        Self((t as u64).into())
+    }
+}
+
+// Reverse TestFelt to Rust types conversion
+
+impl From<TestFelt> for bool {
```

Comment:
Most of these should use `TryFrom`, rather than `From`, since many of these conversions aren't safe. We for sure won't be able to get them merged into `winterfell` with `From` so its better if we stick to what we can get upstreamed (plus, since this is used in our test suite, we avoid accidentally truncating a value and thinking a result is correct when it isn't).

Reply:
Please take a look at my comment above.